artifact/parser.cpp

```diff
@@ -129,6 +136,19 @@ ExpectedArtifact Parse(io::Reader &reader, config::ParserConfig config) {
 		}
 		signature = expected_signature.value();
 		tok = lexer.Next();
+
+		// Verify the signature
+		auto expected_verified = manifest_sig::VerifySignature(
+			*signature, manifest.GetRaw(), config.artifact_verify_keys);
+		if (!expected_verified) {
+			return expected::unexpected(parser_error::MakeError(
+				parser_error::Code::ParseError,
+				"Failed to verify the manifest signature: " + expected_verified.error().message));
+		}
+		if (!expected_verified.value()) {
+			return expected::unexpected(parser_error::MakeError(
+				parser_error::Code::ParseError, "Wrong manifest signature or wrong key"));
```

Comment:
Should it be `ParseError` or a dedicated error for signature?

Reply:
Okey, I created `SignatureVerificationError`